docs/core/examples/ssl_clientauth_client.py

```diff
@@ -8,20 +8,23 @@
 from twisted.python.modules import getModule
 
 
-@defer.inlineCallbacks
-def main(reactor):
+async def main(reactor):
     factory = protocol.Factory.forProtocol(echoclient.EchoClient)
     certData = getModule(__name__).filePath.sibling("public.pem").getContent()
     authData = getModule(__name__).filePath.sibling("server.pem").getContent()
     clientCertificate = ssl.PrivateCertificate.loadPEM(authData)
     authority = ssl.Certificate.loadPEM(certData)
-    options = ssl.optionsForClientTLS("example.com", authority, clientCertificate)
-    endpoint = endpoints.SSL4ClientEndpoint(reactor, "localhost", 8000, options)
-    echoClient = yield endpoint.connect(factory)
-
+    options = ssl.optionsForClientTLS(
+        "example.com",
+        trustRoot=authority,
+        clientCertificate=clientCertificate,
+    )
+    tcpEndpoint = endpoints.HostnameEndpoint(reactor, "localhost", 8000)
+    endpoint = endpoints.wrapClientTLS(options, tcpEndpoint, reactor)
     done = defer.Deferred()
-    echoClient.connectionLost = lambda reason: done.callback(None)
-    yield done
+    echoClient = await endpoint.connect(factory)
+    echoClient.dataReceived = lambda data: done.callback(data)
+    print(await done)
```

Comment:
Maybe just a small comment to explain what is going on here

```suggestion
    # Wait to receive the first data and then end the execution.
    # For this example, we don't do any cleanup or TLS shutdown request.
    print(await done)
```